src/controllers/SizeController.ts

```diff
@@ -63,250 +63,205 @@ export class SizeController
 
-                        if (child.layout && child.layout.style.position)
-                        {
-                            // skip absolute positioned elements
-                            return;
-                        }
+                    if (child.layout && child.layout.style.position)
+                    {
+                        // skip absolute positioned elements
+                        return;
+                    }
 
-                        if (child.layout)
+                    if (child.layout)
+                    {
+                        if (child.layout.style.display === 'block')
                         {
-                            if (child.layout.style.display === 'block')
-                            {
-                                childrenHeight += child.height;
-                            }
-                            else if (child.height > childrenHeight)
-                            {
-                                childrenHeight = child.height;
-                            }
+                            childrenHeight += child.height;
                         }
                         else if (child.height > childrenHeight)
                         {
                             childrenHeight = child.height;
                         }
-                    });
-
-                    if (isItJustAText(this.layout))
+                    }
+                    else if (child.height > childrenHeight)
                     {
-                        finalHeight = this.innerText?.height;
+                        childrenHeight = child.height;
                     }
+                });
 
-                    // height is basing on content height
-                    finalHeight = childrenHeight + paddingTop + paddingBottom;
-
-                    break;
-            }
-        }
-        else
-        {
-            finalHeight = getNumber(height, this.parentHeight);
-        }
-
-        // apply parent paddings
-        if (this.layout.container.parent?.layout && !position)
-        {
-            const { paddingLeft, paddingRight } = this.layout.container.parent?.layout.style;
-
-            const parentPaddingLeft = paddingLeft ?? 0;
-            const parentPaddingRight = paddingRight ?? 0;
-
-            if (this.autoSizeModificator !== 'innerText')
-            {
-                finalWidth -= parentPaddingLeft;
-            }
+                if (isItJustAText(this.layout))
+                {
+                    finalHeight = this.innerText?.height;
+                }
 
-            finalWidth -= parentPaddingRight;
+                // height is basing on content height
+                finalHeight = childrenHeight + paddingTop + paddingBottom;
 
-            this.fitInnerText(finalWidth);
+                break;
 
-            if (isItJustAText(this.layout) && height === 'auto')
-            {
-                finalHeight = this.innerText?.height + paddingBottom + paddingTop;
-            }
```

Comment:
No need of this anymore, paddings applied in places where content size is calculated now